### PR TITLE
Use  inttypes in stm32{f7|h7}, Kinetis, s32k drivers.

### PR DESCRIPTION
**Pull request by davids5**
## Summary

Use the intypes in syslog macros.

## Impact

Allows files to compile.

## Testing

CI

### CODE REVIEW

File: arch/arm/src/stm32f7/stm32_serial.c [hunk 1]
```diff
@@ -3641,10 +3641,10 @@ void arm_serialinit(void)
 
 #if !defined(SERIAL_HAVE_ONLY_DMA)
 #  if defined(SERIAL_HAVE_RXDMA)
-  UNUSED(g_uart_rxdma_ops);
+  UNUSED(&g_uart_rxdma_ops);
```
Comment (antmerlino) [inline arch/arm/src/stm32f7/stm32_serial.c:3644]: @davids5 

Is this change intended to be included? It stands out from everything else.

File: arch/arm/src/stm32f7/stm32_can.c [hunk 2]
```diff
@@ -322,7 +323,7 @@ static uint32_t stm32can_vgetreg(uint32_t addr)
         {
           /* Yes.. then show how many times the value repeated */
 
-          caninfo("[repeats %d more times]\n", count - 3);
+          caninfo("[repeats %" PRId32 " more times]\n", count - 3);
```
Comment (gustavonihei) [inline arch/arm/src/stm32f7/stm32_can.c:326]:
```suggestion
          caninfo("[repeats %" PRIu32 " more times]\n", count - 3);
```
`count` has `uint32_t` type

File: arch/arm/src/stm32f7/stm32_serial.c (hunk 1 above)
Comment (davids5) [reply arch/arm/src/stm32f7/stm32_serial.c:3644]: Yes. it is listed "stm32f7:serial fix compile error from UNUSED() change" 
UNUSED changed recently and broke it.

File: arch/arm/src/stm32f7/stm32_can.c [hunk 3]
```diff
@@ -589,7 +596,7 @@ static void stm32can_reset(FAR struct can_dev_s *dev)
   uint32_t regbit = 0;
   irqstate_t flags;
 
-  caninfo("CAN%d\n", priv->port);
+  caninfo("CAN%" PRId8 "\n", priv->port);
```
Comment (gustavonihei) [inline arch/arm/src/stm32f7/stm32_can.c:599]:
```suggestion
  caninfo("CAN%" PRIu8 "\n", priv->port);
```
`priv->port` has type `uint8_t`.

Comment (gustavonihei) [reply arch/arm/src/stm32f7/stm32_can.c:599]: There are other similar occurrences in this file.

File: arch/arm/src/stm32f7/stm32_can.c [hunk 4]
```diff
@@ -658,15 +665,16 @@ static int stm32can_setup(FAR struct can_dev_s *dev)
   FAR struct stm32_can_s *priv = dev->cd_priv;
   int ret;
 
-  caninfo("CAN%d RX0 irq: %d RX1 irq: %d TX irq: %d\n",
+  caninfo("CAN%" PRId8 " RX0 irq: %" PRId8 " RX1 irq: %" PRId8
+          " TX irq: %" PRId8 "\n",
```
Comment (gustavonihei) [inline arch/arm/src/stm32f7/stm32_can.c:669]:
```suggestion
  caninfo("CAN%" PRIu8 " RX0 irq: %" PRIu8 " RX1 irq: %" PRIu8
          " TX irq: %" PRIu8 "\n",
```
`port`, `canrx` and `cantx` are of type `uint8_t`
```
  uint8_t  port;     /* CAN port number (1 or 2) */
  uint8_t  canrx[2]; /* CAN RX FIFO 0/1 IRQ number */
  uint8_t  cantx;    /* CAN TX IRQ number */
```

File: arch/arm/src/stm32f7/stm32_serial.c (hunk 1 above)
Comment (antmerlino) [reply arch/arm/src/stm32f7/stm32_serial.c:3644]: Cool thanks for the explanation.

File: arch/arm/src/stm32f7/stm32_can.c (hunk 2 above)
Comment (davids5) [reply arch/arm/src/stm32f7/stm32_can.c:326]: @gustavonihei - I was torn on matching the old code usage and fixing them. 

Are you you are siding on the "fixing them" side?

File: arch/arm/src/stm32f7/stm32_can.c (hunk 4 above)
Comment (davids5) [reply arch/arm/src/stm32f7/stm32_can.c:669]: @gustavonihei @antmerlino - Thnak you both!

I will fix these is the AM, squash and force push.

File: arch/arm/src/stm32f7/stm32_can.c [hunk 5]
```diff
@@ -1252,8 +1260,8 @@ static int stm32can_send(FAR struct can_dev_s *dev,
   int dlc;
   int txmb;
 
-  caninfo("CAN%d ID: %d DLC: %d\n",
-          priv->port, msg->cm_hdr.ch_id, msg->cm_hdr.ch_dlc);
+  caninfo("CAN%" PRId8 " ID: %" PRId32 " DLC: %" PRId8 "\n",
+          priv->port, (uint32_t)msg->cm_hdr.ch_id, msg->cm_hdr.ch_dlc);
```
Comment (gustavonihei) [inline arch/arm/src/stm32f7/stm32_can.c:1264]:
```suggestion
  caninfo("CAN%" PRIu8 " ID: %" PRIu32 " DLC: %" PRIu8 "\n",
          priv->port, (uint32_t)msg->cm_hdr.ch_id, msg->cm_hdr.ch_dlc);
```
These variables are all of unsigned types.

File: arch/arm/src/stm32f7/stm32_can.c (hunk 2 above)
Comment (gustavonihei) [reply arch/arm/src/stm32f7/stm32_can.c:326]: Since it is already being changed, I believe it would be best if changed to the correct one :)

Creating another PR just for changing it again seems a waste of resources (review time, CI usage...)

File: arch/arm/src/stm32f7/stm32_can.c (hunk 5 above)
Comment (davids5) [reply arch/arm/src/stm32f7/stm32_can.c]: done in 547e3d7

**Comment by davids5** [comment]
@gustavonihei - I applied you changes ans several more in 547e3d7

**Review by xiaoxiang781216 — APPROVED** [review]
